### PR TITLE
Add namespacing for Redis#scan

### PR DESCRIPTION
**Pull request by jdatti**
Redis SCAN method has been added since 2.8.0
and its support has been added in redis-rb gem since 3.0.6

I do see a similar PR(https://github.com/redis-store/redis-store/pull/314), please let me know if we want to push this after that.

### CODE REVIEW

File: redis-store.gemspec [hunk 1]
```diff
@@ -20,7 +20,7 @@ Gem::Specification.new do |s|
   s.require_paths = ["lib"]
   s.license       = 'MIT'
 
-  s.add_dependency 'redis', '>= 2.2', '< 5'
+  s.add_dependency 'redis', '>= 3.0.6', '< 5'
```
Comment (tubbo) [inline redis-store.gemspec]: can you rebase from master? I updated the redis dependency to require v4 or above

File: lib/redis/store/namespace.rb [hunk 2]
```diff
@@ -39,6 +39,17 @@ def keys(pattern = "*")
         namespace(pattern) { |p| super(p).map { |key| strip_namespace(key) } }
       end
 
+      def scan(cursor, options = {})
+        if options[:match]
+          namespace(options[:match]) do |p|
```
Comment (tubbo) [inline lib/redis/store/namespace.rb]: what does this `p` stand for? can we make the block arg a little more descriptive?

**Comment by tubbo** [comment]
hey @jdatti can you rebase the branch and fix any broken tests?

**Comment by jdatti** [comment]
> hey @jdatti can you rebase the branch and fix any broken tests?

@tubbo Sure let me do that. Thank You.

**Comment by jdatti** [comment]
@tubbo I pushed the changes you've requested, Its ready for the merge.

**Comment by tubbo** [comment]
Thanks @jdatti! Can you give this a whirl in your app by setting `github: "redis-store/redis-store"` as your source in Gemfile?

**Pull request merged**